content/workshops/openshift33/exercise1.10.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Exercise 1.10 - Deploying Java Code on JBoss EAP
+workshops: openshift33
+workshop_weight: 200
+layout: lab
+---
+
+### Background: Source-to-Image (S2I)
+
+In lab four we learned how to deploy a pre-existing Docker image from a Docker
+registry. Now we will expand on that a bit by learning how OpenShift builds a
+Docker images using source code from an existing repository.
+
+[Source-to-Image (S2I)](https://github.com/openshift/source-to-image) is another
+open source project sponsored by Red Hat. Its goal:
+
+```bash
+Source-to-image (S2I) is a tool for building reproducible Docker images. S2I
+produces ready-to-run images by injecting source code into a Docker image and
+assembling a new Docker image which incorporates the builder image and built
+source. The result is then ready to use with docker run. S2I supports
+incremental builds which re-use previously downloaded dependencies, previously
+built artifacts, etc.
+```
+
+OpenShift is S2I-enabled and can use S2I as one of its build mechanisms (in
+addition to building Docker images from Dockerfiles, and "custom" builds).
+
+OpenShift runs the S2I process inside a special *Pod*, called a Build
+Pod, and thus builds are subject to quotas, limits, resource scheduling, and
+other aspects of OpenShift.
+
+A full discussion of S2I is beyond the scope of this class, but you can find
+more information abuot it either in the [OpenShift S2I
+documentation](https://docs.openshift.com/container-platform/3.3/creating_images/s2i.html#overview)
+or on GitHub (following the link above). The only key concept you need to
+remember about S2I is that it's magic.
+
+For a current list of supported runtimes, you can check out the [OpenShift
+Technologies](https://www.openshift.com/container-platform/features.html#technologies) page.
+
+
+### Creating a JBoss EAP application
+
+The sample application that we will be deploying as part of this exercise is
+called `mlbparks`.  This application is a Java EE-based application that
+performs 2D geo-spatial queries against a MongoDB database to locate and map all
+Major League Baseball stadiums in the United States. That was just a fancy way
+of saying that we are going to deploy a map of baseball stadiums.
+
+<blockquote>The first thing you need to do is create a new project called `userXX-mlbparks`: </blockquote>
+
+**Note:** Remember to replace userXX-mlbparks with your correct user number.

```

Comment:
These were corrected.  Without a list of all alert options, I was unaware that 'alert info' was available until this evening.